gui-tk/wiring.tcl

```diff
@@ -809,6 +810,50 @@ proc contextMenu {x y X Y} {
     tk_popup .wiring.context $X $Y
 }
 
+proc setupPickDimMenu {} {
+    global dimLabelPicked
+    if {![winfo exists .wiring.context.pick]} {
+        toplevel .wiring.context.pick
+        wm title .wiring.context.pick "Pick any two dimensions"
+        frame .wiring.context.pick.select
+        scrollbar .wiring.context.pick.select.vscroll -orient vertical -command {
+            .wiring.context.pick.select.lb yview}
+        listbox .wiring.context.pick.select.lb -listvariable dimLabelPicked \
+            -selectmode extended -selectforeground blue \
+            -width 35 \
+            -yscrollcommand {.wiring.context.pick.select.vscroll set} 
+        pack .wiring.context.pick.select.lb -fill both  -expand y -side left
+        pack .wiring.context.pick.select.vscroll -fill y -expand y -side left
+        pack .wiring.context.pick.select
+        buttonBar .wiring.context.pick {
+            set pick {}
+            foreach i [.wiring.context.pick.select.lb curselection] {
+                lappend pick [lindex $dimLabelPicked $i]
+            }
+            if {[lindex $pick 1]==""} {
+				minsky.canvas.item.setDimLabelsPicked [lindex $pick 0] ""
+			} else {
+				minsky.canvas.item.setDimLabelsPicked [lindex $pick 0] [lindex $pick 1]
+			}
+            reset
+        }
+        button .wiring.context.pick.buttonBar.clear -text "Clear" -command {
+            .wiring.context.pick.select.lb selection clear 0 end}
+        pack .wiring.context.pick.buttonBar.clear -side left
+    } else {
+        deiconify .wiring.context.pick
+    }
+        
+    set dimLabelPicked [minsky.canvas.item.dimLabels]
+    for {set i 0} {$i<[llength $dimLabelPicked]} {incr i} {
+        set idx([lindex $dimLabelPicked $i]) $i
+    }
```

Comment:
What is this for?

Reply:
This populates the list that appears in the selection box. Without it, the selection box remains empty.

Reply:
dimLabelPicked is the list of values in the listBox. What is idx? It doesn't appear to be used anywhere I can see.

Reply:
True, I thought you were referring to these lines and the one immediately before it... So, when I commented them out, the list box was empty. These 3 lines contribute nothing. Sorry.